module/friendship/adapter/postgres/repository/user.go

```diff
@@ -2,40 +2,64 @@ package repository
 
 import (
 	"context"
+	"database/sql"
 
 	"github.com/phantranhieunhan/s3-assignment/common"
 	"github.com/phantranhieunhan/s3-assignment/common/adapter/postgres"
 	"github.com/phantranhieunhan/s3-assignment/module/friendship/adapter/postgres/model"
 	"github.com/phantranhieunhan/s3-assignment/module/friendship/domain"
-	"gorm.io/gorm"
+	"github.com/phantranhieunhan/s3-assignment/pkg/util"
+	. "github.com/volatiletech/sqlboiler/v4/queries/qm"
 )
 
 type UserRepository struct {
-	postgres.Database
+	db postgres.Database
 }
 
 func NewUserRepository(db postgres.Database) UserRepository {
 	return UserRepository{
-		Database: db,
+		db: db,
 	}
 }
 
-func (f UserRepository) GetUserIDsByEmails(ctx context.Context, emails []string) (map[string]string, error) {
-	var users model.Users
-	if err := f.Model(ctx).Table(model.User{}.TableName()).Where("email IN ?", emails).Find(&users).Error; err != nil {
-		zResult := make(map[string]string, 0)
-		if err == gorm.ErrRecordNotFound {
-			return zResult, domain.ErrRecordNotFound
+func (f UserRepository) GetUserIDsByEmails(ctx context.Context, emails []string) (map[string]string, []string, error) {
+	users, err := model.Users(AndIn("email IN ?", util.InterfaceSlice(emails)...)).All(ctx, f.db.DB)
+	// users, err := model.Users().All(ctx, f.db.DB)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil, domain.ErrRecordNotFound
 		}
-		return zResult, common.ErrDB(err)
+		return nil, nil, common.ErrDB(err)
 	}
 	if len(users) != len(emails) {
-		return nil, domain.ErrNotFoundUserByEmail
+		return nil, nil, domain.ErrNotFoundUserByEmail
 	}
 	result := make(map[string]string, 0)
+	sliceString := make([]string, 0, len(users))
 	for _, v := range users {
-		result[v.Email] = v.Base.Id
+		result[v.Email] = v.ID
+		sliceString = append(sliceString, v.ID)
 	}
 
-	return result, nil
+	return result, sliceString, nil
```

Comment:
Again, complex function result will later require further handling from source functions to call